File: programmers/기능개발.py
from collections import deque

def solution(progresses, speeds): 
    answer = []
    days = 0
    count = 0
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    while progresses:
        if progresses[0] + (days * speeds[0]) >= 100:
            progresses.popleft()
            speeds.popleft()
            count += 1
        else:
            if count > 0:
                answer.append(count)
                count = 0
            days += 1
    answer.append(count)
    
    return answer

File: programmers/level2/기능개발.cpp
#include <vector>
#include <queue>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> remaining;
    
    for (int i=0; i<progresses.size(); ++i) {
        int days = (int) ceil((100 - progresses[i]) / (double) speeds[i]);
        remaining.push(days);
    }
    
    while (!remaining.empty()) {
        int release = remaining.front();
        remaining.pop();
        int cnt = 1;
        while (!remaining.empty() && remaining.front() <= release) {
            ++cnt;
            remaining.pop();
        }
        answer.push_back(cnt);
    }
    
    return answer;
}